test/app-tap/gh-2717-no-quit-sigint.test.lua

```diff
@@ -0,0 +1,201 @@
+    output = output .. ph:read({timeout = 1.0})
+end
+assert(clock.monotonic() - start_time < deadline, 'deadline is over')
+ph:signal(popen.signal.SIGINT)
+
+while output:find(prompt .. '^C\n---\n...\n\n' .. prompt) == nil and
+        clock.monotonic() - start_time < deadline do
+    local data = ph:read({timeout = 1.0})
+    if data ~= nil then
+        output = output .. data
+    end
+end
+assert(clock.monotonic() - start_time < deadline, 'deadline is over')
+
+test:unlike(ph:info().status.state, popen.state.EXITED,
+            'SIGINT doesn\'t kill tarantool in interactive mode')
+test:like(output, prompt .. '^C\n---\n...\n\n' .. prompt,
+          'Ctrl+C discards the input and calls the new prompt')
+
+ph:shutdown({stdin = true})
+ph:close()
+
+--
+-- gh-2717: testing daemonized tarantool on signaling INT
+-- and nested console output.
+--
+local log_file = fio.abspath('tarantool.log')
+local pid_file = fio.abspath('tarantool.pid')
+local xlog_file = fio.abspath('00000000000000000000.xlog')
+local snap_file = fio.abspath('00000000000000000000.snap')
+local sock = fio.abspath('tarantool.soc')
+
+local user_grant = ' box.schema.user.grant(\'guest\', \'super\')'
+local arg = ' -e \"box.cfg{pid_file=\''
+            .. pid_file .. '\', log=\'' .. log_file .. '\', listen=\'unix/:'
+            .. sock .. '\'}' .. user_grant .. '\"'
+
+start_time = clock.monotonic()
+deadline = 10.0
+
+os.remove(log_file)
+os.remove(pid_file)
+os.remove(xlog_file)
+os.remove(snap_file)
+
+local ph_server = popen.shell(TARANTOOL_PATH .. arg, 'r')
+
+local f = process_timeout.open_with_timeout(log_file, file_open_timeout)
+assert(f, 'error while opening ' .. log_file)
+
+cmd = 'ERRINJ_STDIN_ISATTY=1 ' .. TARANTOOL_PATH .. ' -i 2>&1'
+local ph_client = popen.new({cmd}, {
+    shell = true,
+    setsid = true,
+    group_signal = true,
+    stdout = popen.opts.PIPE,
+    stderr = popen.opts.DEVNULL,
+    stdin = popen.opts.PIPE,
+})
+assert(ph_client, 'the nested console is not up')
+
+ph_client:write('require("console").connect(\'unix/:' .. sock .. '\')\n')
+
+local client_data = ''
+while string.endswith(client_data, 'unix/:' .. sock .. '>') == nil
+        and clock.monotonic() - start_time < deadline do
+    local cur_data = ph_client:read({timeout = 3.0})
+    if cur_data ~= nil and cur_data ~= '' then
+        client_data = client_data .. cur_data
+    end
+end
+assert(clock.monotonic() - start_time < deadline, 'deadline is over')
+
+ph_client:signal(popen.signal.SIGINT)
+test:unlike(ph_client:info().status.state, popen.state.EXITED,
+            'SIGINT doesn\'t kill nested tarantool in interactive mode')
+while string.endswith(client_data, 'C\n---\n...\n\nunix/:' .. sock .. '>') == nil
+        and clock.monotonic() - start_time < deadline do
+    local cur_data = ph_client:read({timeout = 3.0})
+    if cur_data ~= nil and cur_data ~= '' then
+        client_data = client_data .. cur_data
+    end
+end
```

Comment:
Almost all such blocks end with an assertion regarding spent time. Should we assert after this one as well?

Reply:
Fixed.